BlueElectrum.js

```diff
@@ -309,31 +330,19 @@ module.exports.multiGetTransactionByTxid = async function(txids, batchsize, verb
  * @returns {Promise<Promise<*> | Promise<*>>}
  */
 module.exports.waitTillConnected = async function() {
-  let waitTillConnectedInterval = false;
-  let retriesCounter = 0;
-  return new Promise(function(resolve, reject) {
-    waitTillConnectedInterval = setInterval(() => {
-      if (mainConnected) {
-        clearInterval(waitTillConnectedInterval);
-        resolve(true);
-      }
-
-      if (wasConnectedAtLeastOnce && mainClient.status === 1) {
-        clearInterval(waitTillConnectedInterval);
-        mainConnected = true;
-        resolve(true);
-      }
+  const retriesMax = 30;
```

Comment:
Consider using a polling library here, like https://www.npmjs.com/package/promise-poller

Reply:
I rethought the conception now and I think this waitTIllConnected is not needed at all now. I'd remove it from sagas and keep it only for tests purposes